build/helper/metadata_add_all.py

```diff
@@ -201,11 +191,36 @@ def add_all_function_metadata(functions, config):
     return functions
 
 
+def _compare_values(actual, expected):
```

Comment:
What are these changes about? Are they needed for the PR?

Reply:
They help for tracking down why the actual and expected don't match. When `assert actual == expected` there are two huge dictionaries that differ in some small way. This breaks the check down into smaller parts so it can tell you how they don't match. Such and such key is missing or something like that.

Reply:
I think this should be pulled into its own PR.

Reply:
Moved to #410